default/python/universe_generation/starsystems.py

```diff
@@ -59,10 +59,18 @@ def name_planets(system):
         # use different naming methods for "normal" planets and asteroid belts
         if fo.planet_get_type(planet) == fo.planetType.asteroids:
             # get localized text from stringtable
-            name = fo.get_name(system) + " " + fo.user_string("NEW_ASTEROID_SUFFIX_ORBIT_" + (fo.roman_number(min(7, orbit_number))))
+            asteroid_suffix = "NEW_ASTEROID_SUFFIX_ORBIT_" + (fo.roman_number(orbit_number))
+            i = 1
+            while not fo.user_string_exists(asteroid_suffix):
+                if (orbit_number - i) < 1:
+                    asteroid_suffix = "NEW_ASTEROID_SUFFIX_ORBIT_I"
+                    break
+                asteroid_suffix = "NEW_ASTEROID_SUFFIX_ORBIT_" + (fo.roman_number(orbit_number - i))
+                i -= 1
+            name = fo.get_name(system) + " " + fo.user_string(asteroid_suffix)
         else:
             # set name to system name + planet number as roman number...
-            name = fo.get_name(system) + " " + fo.roman_number(min(7, planet_number))
+            name = fo.get_name(system) + " " + fo.roman_number(planet_number)
```

Comment:
what happens if there are more than 7 planets? (thought I already commented about that...?)